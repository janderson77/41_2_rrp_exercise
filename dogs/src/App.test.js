import { render, fireEvent } from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import App from './App';

it('renders without crashing', () => {
  render(
  <MemoryRouter>
    <App />
  </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const {asFragment} = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot();
})

it('works when you click a navbar link', () => {
  const {queryByText, queryAllByText} = render((
    <MemoryRouter>
    <App />
  </MemoryRouter>
  ))

  expect(queryByText('Home')).toBeInTheDocument();
  const link = queryAllByText('Whiskey')
  fireEvent.click(link[0])
  expect(queryByText('Whiskey loves eating popcorn.')).toBeInTheDocument()
})

it('works when you click on the card link', () => {
  const {queryByText, queryAllByText} = render((
    <MemoryRouter>
    <App />
  </MemoryRouter>
  ))

  expect(queryByText('Home')).toBeInTheDocument();
  const link = queryAllByText('Duke')
  fireEvent.click(link[1])
  expect(queryByText('Duke believes that ball is life.')).toBeInTheDocument()
})

it('redirects to /dogs if url is invalid', () => {
  const history = createMemoryHistory()
  history.push('/dogs/aoignweiog')
  const {queryByText} = render((
    <Router history={history}>
    <App />
  </Router>
  ))

  expect(queryByText('Go Back')).not.toBeInTheDocument();
})