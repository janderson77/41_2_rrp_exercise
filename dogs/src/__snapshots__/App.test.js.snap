// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/dogs"
      >
        Home
      </a>
      <a
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
      <a
        href="/dogs/duke"
      >
        Duke
      </a>
      <a
        href="/dogs/perry"
      >
        Perry
      </a>
      <a
        href="/dogs/tubby"
      >
        Tubby
      </a>
    </nav>
    <div
      class="container"
    >
      <div
        class="Doglist"
      >
        <div
          class="DogCard"
        >
          <img
            alt="Whiskey"
            class="DogCard-img"
            src="whiskey.jpg"
          />
          <hr />
          <h3
            class="DogCard-link"
          >
            <a
              href="/dogs/whiskey"
            >
              Whiskey
            </a>
          </h3>
        </div>
        <div
          class="DogCard"
        >
          <img
            alt="Duke"
            class="DogCard-img"
            src="duke.jpg"
          />
          <hr />
          <h3
            class="DogCard-link"
          >
            <a
              href="/dogs/duke"
            >
              Duke
            </a>
          </h3>
        </div>
        <div
          class="DogCard"
        >
          <img
            alt="Perry"
            class="DogCard-img"
            src="perry.jpg"
          />
          <hr />
          <h3
            class="DogCard-link"
          >
            <a
              href="/dogs/perry"
            >
              Perry
            </a>
          </h3>
        </div>
        <div
          class="DogCard"
        >
          <img
            alt="Tubby"
            class="DogCard-img"
            src="tubby.jpg"
          />
          <hr />
          <h3
            class="DogCard-link"
          >
            <a
              href="/dogs/tubby"
            >
              Tubby
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
